app.py

```diff
@@ -0,0 +1,139 @@
+import os
+import json
+import requests
+from flask import Flask, redirect, request
+from typing import Dict, Optional
+from id3c.cli.redcap import is_complete
+
+
+REDCAP_API_TOKEN = os.environ['REDCAP_API_TOKEN']
+REDCAP_API_URL = os.environ['REDCAP_API_URL']
+app = Flask(__name__)
+
+
+def fetch_user_data(net_id: str) -> Optional[Dict[str, str]]:
+    """
+    Exports a REDCap record matching the given *net_id*. Returns None if no
+    match is found.
+
+    Raises an :class:`AssertionError` if REDCap returns multiple matches for the
+    given *net_id*.
+    """
+    fields = [
+        'netid', 'record_id', 'eligibility_screening_complete', 'consent_form_complete',
+        'enrollment_questionnaire_complete'
+    ]
+
+    filter_logic = f'[netid] = "{net_id}"'
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'csvDelimiter': '',
+        'filterLogic': filter_logic,
+        'rawOrLabel': 'raw',
+        'rawOrLabelHeaders': 'raw',
+        'exportCheckboxLabel': 'false',
+        'exportSurveyFields': 'false',
+        'exportDataAccessGroups': 'false',
+        'returnFormat': 'json'
+    }
+    data['fields'] = ",".join(map(str, fields))
+
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+
+    if len(response.json()) == 0:
+        return None
+
+    assert len(response.json()) == 1, \
+        f"Error: Multiple records matching {filter_logic}"
+
+    return response.json()[0]
+
+def register_net_id(net_id: str) -> str:
+    """
+    Returns the REDCap record ID of the participant newly registered under the
+    given *net_id*.
+    """
+    # REDCap enforces that we must provide a non-empty record ID. Because we're
+    # using `forceAutoNumber` in the POST request, we do not need to provide a
+    # real record ID.
+    values = [{'netid': net_id, 'record_id': 'record ID cannot be blank'}]
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'overwriteBehavior': 'normal',
+        'forceAutoNumber': 'true',
+        'data': json.dumps(values),
+        'returnContent': 'ids',
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.json()[0]
+
+
+def generate_survey_link(record_id: str) -> str:
+    """
+    Returns a generated survey link to the eligibility screening instrument for
+    the given *record_id*.
+    """
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'surveyLink',
+        'format': 'json',
+        'instrument': 'eligibility_screening',
+        'event': 'enrollment_arm_1',  # TODO subject to change
```

Comment:
Does it make sense to pull this from the `redcap_event_name` parameter returned by REDCap when we export a record? Currently, in development, the value is 'enrollment_arm_1'.

Reply:
I think this makes sense, this way you don't get all of the repeating event records (which would fail the check of only returning one record per net_id)

Reply:
This makes a lot of sense. You just want the enrollment event, not the repeating encounter events.

Reply:
Ah, but it just occurred to me the `redcap_event_name` parameter won't be available for participants who were just enrolled on this visit to the Husky Musher. Perhaps it's OK to continue to hard-code `enrollment_arm_1` as we've done here?

Reply:
OH, sorry, I completely missed that this is for generating the survey link, not for pulling the record.

Yes, I think we need to hard-code the event name of the enrollment event